Common/include/option_structure.inl

```diff
@@ -378,165 +277,86 @@ public:
 
 public:
-  COptionDoubleList(string option_field_name, unsigned short & list_size, su2double * & option_field) : field(option_field), size(list_size) {
-    this->name = option_field_name;
-  }
+  COptionScalarList() = delete;
 
-  ~COptionDoubleList() override {};
-  string SetValue(vector<string> option_value) override {
-    COptionBase::SetValue(option_value);
-    // The size is the length of option_value
-    unsigned short option_size = option_value.size();
-    if (option_size == 1 && option_value[0].compare("NONE") == 0) {
-      // No options
-      this->size = 0;
-      return "";
-    }
-
-    this->size = option_size;
-
-    // Parse all of the options
-    su2double * vals = new su2double[option_size];
-    for (unsigned long i  = 0; i < option_size; i++) {
-      istringstream is(option_value[i]);
-      su2double val;
-      if (!(is >> val)) {
-        delete [] vals;
-        return badValue(option_value, "su2double list", this->name);
-      }
-      vals[i] = val;
-    }
-    this->field = vals;
-    return "";
-  }
-
-  void SetDefault() override {
-    this->size = 0; // There is no default value for list
+  COptionScalarList(const string& type_name,
+                    const string& option_field_name,
+                    unsigned short& list_size,
+                    Scalar*& option_field) :
+    field(option_field),
+    name(option_field_name),
+    mySize(list_size),
+    typeName(type_name) {
   }
-};
 
-class COptionShortList : public COptionBase {
-  short * & field; // Reference to the feildname
-  string name; // identifier for the option
-  unsigned short & size;
+  ~COptionScalarList() override = default;
 
-public:
-  COptionShortList(string option_field_name, unsigned short & list_size,  short * & option_field) : field(option_field), size(list_size) {
-    this->name = option_field_name;
-  }
-
-  ~COptionShortList() override {};
   string SetValue(vector<string> option_value) override {
     COptionBase::SetValue(option_value);
     // The size is the length of option_value
-    unsigned short option_size = option_value.size();
-    if (option_size == 1 && option_value[0].compare("NONE") == 0) {
+    mySize = option_value.size();
+    if (mySize == 1 && option_value[0].compare("NONE") == 0) {
       // No options
-      this->size = 0;
+      mySize = 0;
       return "";
     }
-    this->size = option_size;
 
     // Parse all of the options
-    short * vals = new  short[option_size];
-    for (unsigned long i  = 0; i < option_size; i++) {
+    field = new Scalar[mySize];
+    for (unsigned short i = 0; i < mySize; i++) {
       istringstream is(option_value[i]);
-      unsigned short val;
+      Scalar val;
       if (!(is >> val)) {
-        delete [] vals;
-        return badValue(option_value, "short", this->name);
+        delete [] field;
+        return badValue(option_value, typeName+" list", name);
       }
-      vals[i] = val;
+      field[i] = std::move(val);
```

Comment:
move instead of copy for that little bit of performance I guess?

Reply:
For when this deals with strings.